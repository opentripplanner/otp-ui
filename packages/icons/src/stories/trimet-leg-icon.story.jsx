import React from "react";

import TrimetLegIcon from "../trimet-leg-icon";

import LegIconRenderer from "./leg-icon-renderer";

export default {
  title: "Icons/TrimetLegIcon",
  component: TrimetLegIcon
};

export const Trimetxamples = () => (
  <LegIconRenderer component={TrimetLegIcon} />
);
