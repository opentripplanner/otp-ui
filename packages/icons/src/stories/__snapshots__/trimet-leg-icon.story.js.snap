// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons/TrimetLegIcon Trimetxamples smoke-test 1`] = `
<table class="icon-renderer__StyledTable-sc-epf3ui-0 fqtAeK">
  <thead>
    <tr>
      <th>
        Leg Type
      </th>
      <th>
        Result
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Personal bike
      </td>
      <td>
        <div>
          <svg viewbox="0 0 390 390"
               leg="[object Object]"
          >
            <path d="M117.1 116.3h33.8c4.7 0 8.4-3.8 8.4-8.4 0-4.7-3.8-8.4-8.4-8.4h-33.8c-4.7 0-8.4 3.8-8.4 8.4-.1 4.6 3.7 8.4 8.4 8.4z">
            </path>
            <path d="M316.4 162.3c-10 0-19.5 2-28.3 5.6-2.5-6.3-4.6-11.9-6-15.8 0 0 0-.1-.1-.2l-24.7-65.8c-1.2-3.3-4.4-5.5-7.9-5.5H217c-4.7 0-8.4 3.8-8.4 8.4 0 4.7 3.8 8.4 8.4 8.4h26.7l10.4 27.9h-113c-3 0-5.6 1.6-7.1 3.9l-25.8 40.6c-9.9-4.8-20.9-7.5-32.6-7.5-41.2 0-74.6 33.4-74.6 74.5 0 41.2 33.4 74.6 74.6 74.6 38.3 0 69.9-28.9 74.1-66.1h30.2c2.4 0 4.7-1 6.2-2.7l81-82.7c1.1 3 3.3 8.8 6.2 16.1-19 13.5-31.4 35.7-31.4 60.8 0 41.2 33.4 74.6 74.6 74.6 41.2 0 74.6-33.4 74.6-74.6-.1-41.1-33.5-74.5-74.7-74.5zm-240.8 83h57.1c-1.8 12.2-7.4 23.5-16.3 32.4-10.9 10.9-25.4 16.9-40.8 16.9-15.4 0-29.9-6-40.8-16.9-10.9-10.9-16.9-25.4-16.9-40.8 0-15.4 6-29.9 16.9-40.8 10.9-10.9 25.4-16.9 40.8-16.9 15.4 0 29.9 6 40.8 16.9 8.9 8.9 14.5 20.1 16.3 32.4H75.6c-4.7 0-8.4 3.8-8.4 8.4-.1 4.6 3.7 8.4 8.4 8.4zm74.1-16.9c-2.3-20-12.4-37.6-27.2-49.6l16.7-26.3 28.5 75.9h-18zm33.4-6.9l-29.8-79.3h107.2l.1.2-77.5 79.1zm174.1 56.1c-10.9 10.9-25.4 16.9-40.8 16.9-15.4 0-29.9-6-40.8-16.9-10.9-10.9-16.9-25.4-16.9-40.8 0-15.4 6-29.9 16.9-40.8 1.4-1.4 2.9-2.8 4.4-4 0 .1 0 .1.1.2 11.1 25.1 21.4 42.2 30.5 50.8 1.6 1.5 3.7 2.3 5.8 2.3 2.2 0 4.5-.9 6.1-2.6 3.2-3.4 3-8.7-.3-11.9-9-8.5-19.4-28.8-27.5-47.3 6.8-2.8 14.2-4.2 21.7-4.2 15.4 0 29.9 6 40.8 16.9 10.9 10.9 16.9 25.4 16.9 40.8 0 15.3-6 29.7-16.9 40.6z">
            </path>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Bike rental
      </td>
      <td>
        <div>
          <svg xmlns="http://www.w3.org/2000/svg"
               viewbox="0 0 418.62 160.29"
               height="100%"
               width="100%"
          >
            <path d="M230.91,500.07c-6.66,1.79-12,1.53-15.08-.71-7.84-5.72-1.89-17.25.27-20.68a103,103,0,0,0-10.64,13c-4.68,7.2-6.68,15.07-3.52,20.13,5.69,9.12,19.32,4.91,28,1.25l80.66-34Z"
                  transform="translate(-37 -357)"
            >
            </path>
            <path d="M37,431.27V357H61.88q14,0,20.68,5.77a19.31,19.31,0,0,1,6.83,15.29q0,14-14.78,15.11Q83,394.33,87,398.37T90.87,410q0,9.58-6.62,15.39T67,431.27H37Zm21.4-46h1.44a9.84,9.84,0,0,0,6.08-1.68,5.57,5.57,0,0,0,2.23-4.71q0-6.44-7.54-6.45h-0.9l-1.31.07v12.78h0Zm0,30.25,1.08,0.08h0.77a9.55,9.55,0,0,0,6.39-2A6.76,6.76,0,0,0,69,408.18a6.47,6.47,0,0,0-2.38-5.33,10.12,10.12,0,0,0-6.55-1.93H58.4v14.61h0Z"
                  transform="translate(-37 -357)"
            >
            </path>
            <rect x="56.7"
                  width="21.4"
                  height="74.27"
            >
            </rect>
            <polygon points="141 74.27 117.92 74.27 103.7 40.02 104.27 74.27 82.87 74.27 82.87 0 104.27 0 103.7 31.84 116.33 0 138.9 0 121.81 35.84 141 74.27">
            </polygon>
            <path d="M315.73,394.39q0,16.4-8.26,26.66a26.16,26.16,0,0,1-21.39,10.22A28.29,28.29,0,0,1,266,423.42q-10.5-9.94-10.5-29.82,0-19,11-29.12A27.67,27.67,0,0,1,285.82,357q13.7,0,21.81,10.12t8.11,27.26m-22.22.17q0-19.76-7.59-19.76-3.79,0-6.05,5.19-2.21,4.81-2.21,14.15t2.08,14.27q2.08,5.07,5.77,5.07t5.85-5q2.15-5,2.15-13.91"
                  transform="translate(-37 -357)"
            >
            </path>
            <path d="M383.26,431.27h-20.7l-5-26q-1.45-7.55-2.18-13.08-0.36-2.74-.62-4.57c-0.18-1.22-.29-2.15-0.36-2.79q-0.26,2-.77,7.36-0.24,2.74-.56,5.77t-0.8,6.4l-3.71,26.93h-20L310.92,357h21.14l3.49,20.44q2,12.22,2.77,19.19l0.68,6.79,0.67-5.82q1-10.2,2.68-19.39L346.24,357H364l3.47,17.85q2.53,12.45,4.35,29.48,0.57-7.65,1.23-13.56T374.54,380l3.9-23h20.88Z"
                  transform="translate(-37 -357)"
            >
            </path>
            <path d="M399.69,431.27V357H420.1l10.73,26.07,2,5.29q1.13,3,2.44,6.88l2.21,6.54q-1.18-9.67-1.79-16.55T435,373.69V357h20.62v74.27H435L423.61,404q-1.8-4.38-3.24-8.2t-2.53-7.29q1.28,7.84,1.87,13.66a101.82,101.82,0,0,1,.59,10.2v19H399.69v-0.05Z"
                  transform="translate(-37 -357)"
            >
            </path>
            <polygon points="140.86 0 140.86 74.27 180.62 74.27 180.62 56.47 162.25 56.47 162.25 45.07 178.72 45.07 178.72 27.27 162.25 27.27 162.25 17.8 189.71 17.8 189.71 74.27 211.1 74.27 211.1 17.8 223.37 17.8 223.37 0 140.86 0">
            </polygon>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Bus
      </td>
      <td>
        <div>
          <svg viewbox="0 0 390 390"
               leg="[object Object]"
          >
            <path d="m 325.28866,67.290913 c 0,-31.901057 -58.52714,-48.981938 -130.6185,-48.981938 -72.09137,0 -130.492908,17.080881 -130.618502,48.981938 L 56.515975,211.72483 v 90.4282 c 0,5.52615 4.521409,10.04757 10.047577,10.04757 h 8.79163 v 25.11894 c 0,8.28925 6.782114,15.07137 15.071365,15.07137 h 10.047583 c 8.28925,0 15.07136,-6.78212 15.07136,-15.07137 V 312.2006 h 158.24934 v 25.11894 c 0,8.28925 6.78211,15.07137 15.07136,15.07137 h 10.04759 c 8.28925,0 15.07136,-6.78212 15.07136,-15.07137 V 312.2006 h 8.79162 c 5.52617,0 10.04758,-4.52142 10.04758,-10.04757 v -90.4282 z M 148.82809,38.404129 h 91.68414 c 4.89819,0 8.79163,3.893436 8.79163,8.79163 0,4.898194 -3.89344,8.79163 -8.79163,8.79163 h -91.68414 c -4.89819,0 -8.79163,-3.893436 -8.79163,-8.79163 0,-4.898194 3.89344,-8.79163 8.79163,-8.79163 z M 95.450336,269.49839 c -10.424362,0 -18.839207,-8.41483 -18.839207,-18.8392 0,-10.42437 8.414845,-18.83921 18.839207,-18.83921 10.424364,0 18.839204,8.41484 18.839204,18.83921 0,10.42437 -8.41484,18.8392 -18.839204,18.8392 z m 198.439644,0 c -10.42436,0 -18.8392,-8.41483 -18.8392,-18.8392 0,-10.42437 8.41484,-18.83921 18.8392,-18.83921 10.42437,0 18.83921,8.41484 18.83921,18.83921 0,10.42437 -8.41484,18.8392 -18.83921,18.8392 z m 8.16366,-57.77356 H 87.286679 c -6.530924,0 -11.680308,-5.40057 -11.303524,-11.9315 L 82.137296,83.115846 C 82.388485,77.715274 86.7843,73.570648 92.184873,73.570648 H 296.02509 c 6.02855,0 10.92674,4.647004 11.30352,10.675551 l 6.02855,115.547131 c 0.37679,6.53093 -4.77259,11.9315 -11.30352,11.9315 z">
            </path>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Micromobility (from Razor company)
      </td>
      <td>
        <div>
          <svg viewbox="0 0 512 512"
               leg="[object Object]"
          >
            <path d="M464.027 393.21L416.602 51.747C412.504 22.246 386.953 0 357.172 0H289v40h68.172c9.93 0 18.441 7.414 19.808 17.246l36.149 260.25C361 332.586 321.293 377.22 313.426 432H116.559c-8.254-23.277-30.485-40-56.559-40-33.086 0-60 26.914-60 60s26.914 60 60 60c26.074 0 48.305-16.723 56.559-40H352v-20c0-43.473 27.887-80.55 66.71-94.297l5.7 41.043C405.176 408.75 392 428.86 392 452c0 33.086 26.914 60 60 60s60-26.914 60-60c0-28.965-20.633-53.203-47.973-58.79zM60 472c-11.027 0-20-8.973-20-20s8.973-20 20-20 20 8.973 20 20-8.973 20-20 20zm392 0c-11.027 0-20-8.973-20-20s8.973-20 20-20 20 8.973 20 20-8.973 20-20 20zm0 0">
            </path>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Park &amp; Ride
      </td>
      <td>
        <div>
          <svg viewbox="0 0 390 390"
               leg="[object Object]"
          >
            <path d="M371.2 175.3l-3.7-12.1-29.8-95.7c-5.8-18.6-22.9-31.2-42.4-31.2H94.7c-19.5 0-36.5 12.6-42.4 31.2l-29.8 95.7-3.7 12.1C7.6 181.4 0 193.1 0 206.5v76.2c0 7.8 6.4 14.2 14.2 14.2h12.4v35.5c0 11.7 9.6 21.3 21.3 21.3H62c11.7 0 21.3-9.6 21.3-21.3v-35.5h223.4v35.5c0 11.7 9.6 21.3 21.3 21.3h14.2c11.7 0 21.3-9.6 21.3-21.3v-35.5h12.4c7.8 0 14.2-6.4 14.2-14.2v-76.2c-.1-13.3-7.7-25.1-18.9-31.2zM55 252.8c-14.7 0-26.6-11.9-26.6-26.6s11.9-26.6 26.6-26.6 26.6 11.9 26.6 26.6-11.9 26.6-26.6 26.6zm240.7-81.6c-9-14.9-25.4-24.8-44-24.8s-34.9 9.9-44 24.8H47.9l29.8-95.6c2.3-7.4 9.2-12.4 16.8-12.4h200.9c7.8 0 14.5 5 16.8 12.4l29.8 95.7-46.3-.1zm39.3 81.6c-14.7 0-26.6-11.9-26.6-26.6s11.9-26.6 26.6-26.6c14.7 0 26.6 11.9 26.6 26.6s-11.8 26.6-26.6 26.6z">
            </path>
            <circle cx="251.7"
                    cy="107.4"
                    r="33.7"
            >
            </circle>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        TNC (Uber)
      </td>
      <td>
        <div>
          <svg viewbox="0 0 390 390"
               leg="[object Object]"
          >
            <path d="M371.2 175.3l-3.7-12.1-29.8-95.7c-5.8-18.6-22.9-31.2-42.4-31.2H94.7c-19.5 0-36.5 12.6-42.4 31.2l-29.8 95.7-3.7 12.1C7.6 181.4 0 193.1 0 206.5v76.2c0 7.8 6.4 14.2 14.2 14.2h12.4v35.5c0 11.7 9.6 21.3 21.3 21.3H62c11.7 0 21.3-9.6 21.3-21.3v-35.5h223.4v35.5c0 11.7 9.6 21.3 21.3 21.3h14.2c11.7 0 21.3-9.6 21.3-21.3v-35.5h12.4c7.8 0 14.2-6.4 14.2-14.2v-76.2c-.1-13.3-7.7-25.1-18.9-31.2zM55 252.8c-14.7 0-26.6-11.9-26.6-26.6s11.9-26.6 26.6-26.6 26.6 11.9 26.6 26.6-11.9 26.6-26.6 26.6zm240.7-81.6c-9-14.9-25.4-24.8-44-24.8s-34.9 9.9-44 24.8H47.9l29.8-95.6c2.3-7.4 9.2-12.4 16.8-12.4h200.9c7.8 0 14.5 5 16.8 12.4l29.8 95.7-46.3-.1zm39.3 81.6c-14.7 0-26.6-11.9-26.6-26.6s11.9-26.6 26.6-26.6c14.7 0 26.6 11.9 26.6 26.6s-11.8 26.6-26.6 26.6z">
            </path>
            <circle cx="251.7"
                    cy="107.4"
                    r="33.7"
            >
            </circle>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Tram
      </td>
      <td>
        <div>
          <svg viewbox="0 0 390 390"
               leg="[object Object]"
          >
            <path d="M257.4 300h7.6c16.6 0 30-13.4 30-30V92c0-16.6-13.4-30-30-30h-56.5l40.3-29.9 11.6 9.9c.8.6 1.7 1 2.6 1 1.1 0 2.2-.5 3-1.4 1.4-1.7 1.2-4.2-.4-5.6l-14-12c-.7-.6-1.6-1-2.6-1H141c-1 0-1.9.3-2.6 1l-14 12c-1.7 1.4-1.9 4-.4 5.6 1.4 1.7 4 1.9 5.6.4l11.6-9.9L181.5 62H125c-16.6 0-30 13.4-30 30v178c0 16.6 13.4 30 30 30h7.6L84 372h26l10.8-16h148.4l10.8 16h26l-48.6-72zm6.6-34c-8.3 0-15-6.7-15-15s6.7-15 15-15 15 6.7 15 15-6.7 15-15 15zM153.1 31h83.8L195 62l-41.9-31zm5.4 45h73c3.9 0 7 3.1 7 7s-3.1 7-7 7h-73c-3.9 0-7-3.1-7-7s3.1-7 7-7zM111 112c0-4.4 3.6-8 8-8h152c4.4 0 8 3.6 8 8v100c0 4.4-3.6 8-8 8H119c-4.4 0-8-3.6-8-8V112zm131.2 204h-94.4l10.8-16h72.8l10.8 16zM126 266c-8.3 0-15-6.7-15-15s6.7-15 15-15 15 6.7 15 15-6.7 15-15 15zm2.9 78l10.8-16h110.6l10.8 16H128.9z">
            </path>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Walk
      </td>
      <td>
        <div>
          <svg viewbox="0 0 390 390"
               leg="[object Object]"
          >
            <path d="m 207.60225,19.503527 c 16.02,0 29.07,13.05 29.07,29.07 0,16.020001 -13.05,29.07 -29.07,29.07 -16.02,0 -29.07,-13.049999 -29.07,-29.07 0,-16.02 13.05,-29.07 29.07,-29.07 z m 88.65,181.620003 c -4.05,5.94 -12.15,7.47 -18.09,3.42 0,0 -40.23,-27.45 -43.11,-29.43 -2.88,-1.98 -4.59,-6.12 -5.85,-8.82 -0.9,-1.8 -5.4,-10.62 -8.37,-16.47 l -3.6,15.3 -9.36,42.39 c 0,0 50.13,59.04 51.12,60.21 1.44,1.62 2.25,4.95 2.88,7.47 0.63,2.43 13.41,71.91 13.41,71.91 1.98,10.71 -5.13,21.06 -15.84,23.13 -10.71,1.98001 -21.06,-5.13 -23.13,-15.84 l -12.78,-68.67 -41.67,-45.54 c 0,0 -9.27,43.29 -9.45,44.19 -0.18,0.9 -0.63,2.25 -1.17,3.51 -0.54,1.17 -42.84,72.99 -42.84,72.99 -5.58,9.36 -17.73,12.51 -27.18,6.93 -9.359998,-5.58 -12.509998,-17.73 -6.929998,-27.18 l 37.619998,-63.27 30.24,-136.89 -20.34,16.2 -11.25,49.59 c 0,7.02 -7.92,11.61 -14.94,10.08 -7.02,-1.53 -11.16,-4.41 -9.54,-15.39 h 0.18 c 0,0 11.7,-55.08 12.6,-57.51 0.81,-2.43 2.07,-5.13 3.33,-6.12 8.46,-6.84 41.22,-33.66 50.85,-41.400003 11.07,-9.000001 15.93,-12.329999 25.38,-12.329999 7.11,0 14.85,2.159999 23.04,10.62 3.6,3.690001 6.21,9.720002 8.01,13.500002 1.8,3.69 24.39,48.33 24.39,48.33 l 38.97,27 c 5.94,4.05 7.47,12.15 3.42,18.09 z">
            </path>
          </svg>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
